
---
title: "Gut microbiome and giant liver fluke infections among red deer in the Bohemian Forest Ecosystem"
author: "Ramona Fleischer"
date: 'r Sys.Date()'
output:
  html_document:
    code_folding: show
    fig_caption: yes
    fig_height: 10
    fig_width: 10
    toc: yes
    toc_depth: 4
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '4'
  word_document:
    toc: yes
    toc_depth: '4'
editor_options:
  chunk_output_type: inline
---

```{r}
setwd("D:\\Ramona_F\\8Deer Project\\R\\deer_script_published")
```

```{r}
library(phyloseq)
library(tidyverse)
library(vegan)
library(ggplot2)
library(ggpubr)
library(decontam)
library(effects)
library(picante)
library(microbiome)
library(scales)
library(ggrepel)
library(MuMIn)
library(dplyr)
library(tibble)
```

We will run 1) a model including ALL variables on ALL individuals. 

And 2) a model including only the INFECTED individuals to examine the effect of F_Count on gut microbial diversity.

We work with the filtered phyloseq here (N=495)

Import data

```{r}
#deer_all <- readRDS("all.deer.filt.RDS")
```

Organize metadata

```{r}
#str(sample_data(deer_all))

#sample_data(deer_all)$F_Count <- as.numeric(sample_data(deer_all)$F_Count)
#sample_data(deer_all)$Fluke <- as.factor(sample_data(deer_all)$Fluke)
#sample_data(deer_all)$HuntingMonth <- as.factor(sample_data(deer_all)$HuntingMonth)

#levels(sample_data(deer_all)$HuntingMonth)
#levels(sample_data(deer_all)$HuntingMonth) <- c("April", "August", "December", "February", "January", "July", "June", "June", "March", "May", "November" ,"October", "September") 

#sample_data(deer_all)$HuntingMonth <- factor(sample_data(deer_all)$HuntingMonth, levels = c("January", "February", "March", "April",  "May","June","July","August", "September","October","November" ,"December"))
```

Remove the months February - July

```{r}
#deer_winter <- subset_samples(deer_all, HuntingMonth != "February")

#deer_winter <- subset_samples(deer_winter, HuntingMonth != "March")
#deer_winter <- subset_samples(deer_winter, HuntingMonth != "April")
#deer_winter <- subset_samples(deer_winter, HuntingMonth != "May")
#deer_winter <- subset_samples(deer_winter, HuntingMonth != "June")
#deer_winter <- subset_samples(deer_winter, HuntingMonth != "July")

#deer_winter # 434 samples
#saveRDS(deer_winter, "deer_winter.RDS")
```

```{r}
deer_winter <- readRDS("deer_winter.RDS")
```

Overview sample sizes

Extract metadata

```{r}
meta_winter <- data.frame(sample_data(deer_winter))
```

Check dataset completeness

```{r}
table(meta(meta_winter)$Fluke, useNA = "always") # 16 individuals with NA at Fluke information
table(meta(meta_winter)$Sex, useNA = "always")   # 1 individual with NA at Sex
table(meta(meta_winter)$Age, useNA = "always")   # 0 individuals with NA at Age
table(meta(meta_winter)$Site_2, useNA = "always")   # 0 individuals with NA at Site
table(meta(meta_winter)$HuntingMonth, useNA = "always")   # 0 individuals with NA at Site
```

Fluke prevalence

# including the NA individuals

```{r}
fluke <- ggplot(meta_winter, aes(x=Fluke, fill = Fluke)) +     
  geom_bar(color="black")+
  scale_fill_manual(labels=c("absent", "present", "NA"), values=c("#98A092","#F14747"), na.value = "#EEF0F2")+
  ylab("Individuals [n]")+
  theme_classic(base_size=14)+
  theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())

fluke.sex <- ggplot(meta_winter, aes(x=Sex, fill = Fluke)) +     
  geom_bar(color="black")+
  scale_fill_manual(labels=c("absent", "present", "NA"), values=c("#98A092","#F14747"), na.value = "#EEF0F2")+
  ylab("Individuals [n]")+
  theme_classic(base_size=14)+
  theme(axis.title.x=element_blank())

fluke.age <- ggplot(meta_winter, aes(x=Age_Estimate, fill = Fluke)) +     
  geom_bar(color="black")+
  scale_fill_manual(labels=c("absent", "present", "NA"), values=c("#98A092","#F14747"), na.value = "#EEF0F2")+
  ylab(" ")+
  theme_classic(base_size=14)+
  theme(axis.title.x=element_blank())

meta_winter$Site_2 <- factor(meta_winter$Site_2, levels = c("Bavarian_Forest","Neureichenau","Sumava"))

table(meta_winter$Site_2, meta_winter$Fluke)

fluke.area <- ggplot(meta_winter, aes(x=Site_2, fill = Fluke)) +     
  geom_bar(color="black")+
  scale_fill_manual(labels=c("absent", "present", "NA"), values=c("#98A092","#F14747"), na.value = "#EEF0F2")+
  ylab(" ")+
  theme_classic(base_size=14)+
  theme(axis.title.x=element_blank())+ 
  scale_x_discrete(labels=c("Bavarian_Forest" = "BF-NP", "Neureichenau" = "N-FE", "Sumava" = "S-NP")) # order sites

fluke.season <- ggplot(meta_winter, aes(x=HuntingMonth, fill = Fluke)) +     
  geom_bar(color="black")+
  scale_fill_manual(labels=c("absent", "present", "NA") ,values=c("#98A092","#F14747","#EEF0F2"))+
  ylab(" ")+
  theme_classic(base_size=14)+
  theme(axis.title.x=element_blank())+ 
  scale_x_discrete(labels=c("January" = "Jan", "August" = "Aug", "September" = "Sep", "October" = "Oct", "November" = "Nov", "December" = "Dec")) # order months
```

Remove the individuals without known fluke infection state (N=16)

```{r}
meta_winter1 <- subset(meta_winter, Fluke != "NA")
table(meta_winter)
table(meta_winter1)
```

For Table 1

```{r}
table(meta_winter1$Fluke, meta_winter1$Age_Estimate, useNA = "always")
table(meta_winter1$Fluke, meta_winter1$Sex, useNA = "always")
table(meta_winter1$Fluke, meta_winter1$Site_2, useNA = "always")
table(meta_winter1$Fluke, meta_winter1$HuntingMonth, useNA = "always")
```

# original levels
```{r}
meta_winter1$Age_Estimate <- factor(meta_winter1$Age_Estimate, levels = c("Adult","Subadult","Calf"))
meta_winter1$Site_2 <- factor(meta_winter1$Site_2, levels = c("Bavarian_Forest","Sumava","Neureichenau"))
meta_winter1$HuntingMonth <- factor(meta_winter1$HuntingMonth, levels = c("January","August","September","October","November","December"))
```

Fluke prevalence

# excluding the NA individuals

```{r}
fluke <- ggplot(meta_winter1, aes(x=Fluke, fill = Fluke)) +     
  geom_bar(color="black")+
  scale_fill_manual(labels=c("absent", "present", "NA"), values=c("#98A092","#F14747"), na.value = "#EEF0F2")+
  ylab("Individuals [n]")+
  theme_classic(base_size=14)+
  theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())

fluke.sex <- ggplot(meta_winter1, aes(x=Sex, fill = Fluke)) +     
  geom_bar(color="black")+
  scale_fill_manual(labels=c("absent", "present", "NA"), values=c("#98A092","#F14747"), na.value = "#EEF0F2")+
  ylab("Individuals [n]")+
  theme_classic(base_size=14)+
  theme(axis.title.x=element_blank())+
  scale_x_discrete(labels=c("Female" = "F", "Male" = "M", "NA" = "NA"))

fluke.age <- ggplot(meta_winter1, aes(x=Age_Estimate, fill = Fluke)) +     
  geom_bar(color="black")+
  scale_fill_manual(labels=c("absent", "present", "NA"), values=c("#98A092","#F14747"), na.value = "#EEF0F2")+
  ylab(" ")+
  theme_classic(base_size=14)+
  theme(axis.title.x=element_blank())+
  scale_x_discrete(labels=c("Adult" = "A", "Subadult" = "SA", "Calf" = "C"))

meta_winter1$Site_2 <- factor(meta_winter1$Site_2, levels = c("Bavarian_Forest","Neureichenau","Sumava"))

table(meta_winter1$Site_2, meta_winter1$Fluke)

fluke.area <- ggplot(meta_winter1, aes(x=Site_2, fill = Fluke)) +     
  geom_bar(color="black")+
  scale_fill_manual(labels=c("absent", "present", "NA"), values=c("#98A092","#F14747"), na.value = "#EEF0F2")+
  ylab(" ")+
  theme_classic(base_size=14)+
  theme(axis.title.x=element_blank())+ 
  scale_x_discrete(labels=c("Bavarian_Forest" = "BF-NP", "Neureichenau" = "N-FE", "Sumava" = "S-NP")) # order sites

fluke.season <- ggplot(meta_winter1, aes(x=HuntingMonth, fill = Fluke)) +     
  geom_bar(color="black")+
  scale_fill_manual(labels=c("absent", "present", "NA") ,values=c("#98A092","#F14747","#EEF0F2"))+
  ylab(" ")+
  theme_classic(base_size=14)+
  theme(axis.title.x=element_blank())+ 
  scale_x_discrete(labels=c("January" = "Jan", "August" = "Aug", "September" = "Sep", "October" = "Oct", "November" = "Nov", "December" = "Dec")) # order months
```

Does fluke prevalence differ between sexes, age groups, sites or seasons?

```{r}
# original levels
meta_winter1$Age_Estimate <- factor(meta_winter1$Age_Estimate, levels = c("Subadult","Calf","Adult"))
meta_winter1$Site_2 <- factor(meta_winter1$Site_2, levels = c("Bavarian_Forest","Sumava","Neureichenau"))
meta_winter1$HuntingMonth <- factor(meta_winter1$HuntingMonth, levels = c("January","August","September","October","November","December"))

summary(glm(Fluke~Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_winter1, family = "binomial"))

# relevel
meta_winter1$Age_Estimate <- factor(meta_winter1$Age_Estimate, levels = c("Adult", "Subadult","Calf"))
meta_winter1$Site_2 <- factor(meta_winter1$Site_2, levels = c("Sumava","Neureichenau","Bavarian_Forest"))
meta_winter1$HuntingMonth <- factor(meta_winter1$HuntingMonth, levels = c("August","September","October","November","December","January"))

summary(glm(Fluke~Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_winter1, family = "binomial"))

# relevel
meta_winter1$HuntingMonth <- factor(meta_winter1$HuntingMonth, levels = c("September","October","November","December","January","August"))

summary(glm(Fluke~Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_winter1, family = "binomial"))

# relevel
meta_winter1$HuntingMonth <- factor(meta_winter1$HuntingMonth, levels = c("October","November","December","January","August","September"))

summary(glm(Fluke~Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_winter1, family = "binomial"))

# relevel
meta_winter1$HuntingMonth <- factor(meta_winter1$HuntingMonth, levels = c("November","December","January","August","September","October"))

summary(glm(Fluke~Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_winter1, family = "binomial"))
```

Relevel original

```{r}
meta_winter1$Age_Estimate <- factor(meta_winter1$Age_Estimate, levels = c("Subadult","Calf","Adult"))
meta_winter1$Site_2 <- factor(meta_winter1$Site_2, levels = c("Bavarian_Forest","Sumava","Neureichenau"))
meta_winter1$HuntingMonth <- factor(meta_winter1$HuntingMonth, levels = c("January","August","September","October","November","December"))
```


Fluke infection intensity

```{r}
deer_pos <- subset_samples(deer_winter, Fluke =="1")

table(sample_data(deer_pos)$F_Count)

min(sample_data(deer_pos)$F_Count)
max(sample_data(deer_pos)$F_Count)
mean(sample_data(deer_pos)$F_Count)
median(sample_data(deer_pos)$F_Count)
```

```{r}
tbl <- with(sample_data(deer_pos), table(F_Count))
tbl <- as.data.frame(tbl)

tbl$F_Count <- as.numeric(as.character(tbl$F_Count))

fluke.count <- ggplot(as.data.frame(tbl), aes(as.numeric(as.character(F_Count)), Freq, fill = F_Count)) +     
  geom_bar(color="black", stat="identity")+
  scale_fill_gradient(low = "#FFF0F3",high = "#F14747",space = "Lab",na.value = "grey50",guide = "colourbar",aesthetics = "fill")+
  ylab("Individuals [n]")+
  xlab(" ")+
  theme_classic(base_size=14)
```

Figure 1 b-d

```{r}
Figure1bc <- ggarrange(fluke, fluke.area, common.legend = TRUE, legend = "right", labels = c("b", "c"))
Figure1 <- ggarrange(Figure1bc, fluke.count, legend = "right", nrow=2, labels = c(" ", "d"))
Figure1
```

Supplementary Figure

```{r}
SupplFig2 <- ggarrange(fluke.sex, fluke.age, fluke.season, ncol=3, common.legend = TRUE, legend = "right", labels = c("a","b", "c"))
SupplFig2
```

Does fluke infection intensity differ between sexes, age groups, sites or seasons?

Only within the infected individuals (N=62)

```{r}
meta_pos <- data.frame(sample_data(deer_pos))

hist(meta_pos$F_Count)
hist(log(meta_pos$F_Count))

# original levels
meta_pos$Age_Estimate <- factor(meta_pos$Age_Estimate, levels = c("Subadult","Calf","Adult"))
meta_pos$Site_2 <- factor(meta_pos$Site_2, levels = c("Bavarian_Forest","Sumava","Neureichenau"))
meta_pos$HuntingMonth <- factor(meta_pos$HuntingMonth, levels = c("January","August","September","October","November","December"))

#m <- glm(F_Count ~ Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_pos, family = Gamma("inverse"))
#plot(m)

summary(glm(F_Count ~ Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_pos, family = Gamma("inverse")))
#summary(glm(log(F_Count) ~ Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_pos, family = Gamma("inverse")))

# relevel
meta_pos$Age_Estimate <- factor(meta_pos$Age_Estimate, levels = c("Adult", "Subadult","Calf"))
meta_pos$Site_2 <- factor(meta_pos$Site_2, levels = c("Sumava","Neureichenau","Bavarian_Forest"))
meta_pos$HuntingMonth <- factor(meta_pos$HuntingMonth, levels = c("August","September","October","November","December","January"))

summary(glm(F_Count ~ Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_pos, family = Gamma("inverse")))

# relevel
meta_pos$HuntingMonth <- factor(meta_pos$HuntingMonth, levels = c("September","October","November","December","January","August"))

summary(glm(F_Count ~ Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_pos, family = Gamma("inverse")))

# relevel
meta_pos$HuntingMonth <- factor(meta_pos$HuntingMonth, levels = c("October","November","December","January","August","September"))

summary(glm(F_Count ~ Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_pos, family = Gamma("inverse")))

# relevel
meta_pos$HuntingMonth <- factor(meta_pos$HuntingMonth, levels = c("November","December","January","August","September","October"))

summary(glm(F_Count ~ Sex + Site_2 + Age_Estimate + HuntingMonth, data = meta_pos, family = Gamma("inverse")))
```

# Alpha diversity

1) Number of observed ASVs = microbial richness

2) Shannon Index = microbial eveness

3) Faith's PD = microbial phylogenetic diversity

```{r}
sample_data(deer_winter)$Observed<-estimate_richness(deer_winter, measures = "Observed")$Observed
sample_data(deer_winter)$Shannon<-estimate_richness(deer_winter, measures = "Shannon")$Shannon
sample_data(deer_winter)$Simpson<-estimate_richness(deer_winter, measures = "Simpson")$Simpson

otu.table <- as.data.frame(otu_table(deer_winter))
phylo.meta <- as(sample_data(deer_winter), "data.frame")

df.pd <- pd(t(otu.table), phy_tree(deer_winter),include.root=T) 

sample_data(deer_winter)$PD <- df.pd$PD 
```

# Alpha diversity graphs

```{r}
str(sample_data(deer_winter)$Fluke)

P1 <- ggplot(data=subset(sample_data(deer_winter), !is.na(Fluke)), aes(x = Fluke, y = Observed, color = Fluke, fill = Fluke))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(labels = c("Absent (0)", "Present (1)"),values=c("#98A092","#F14747"))+
  scale_fill_manual(labels = c("Absent (0)", "Present (1)"),values=c("#98A092","#F14747"))+
  geom_point(aes(col = Fluke))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Number of ASVs")+
  xlab("")+ theme(legend.position = "none")

P2 <- ggplot(data=subset(sample_data(deer_winter), !is.na(Fluke)), aes(x = Fluke, y = Shannon, color = Fluke, fill = Fluke))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(labels = c("Absent (0)", "Present (1)"),values=c("#98A092","#F14747"))+
  scale_fill_manual(labels = c("Absent (0)", "Present (1)"),values=c("#98A092","#F14747"))+
  geom_point(aes(col = Fluke))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Shannon Index")+
  xlab("")+ theme(legend.position = "none")

P3 <- ggplot(data=subset(sample_data(deer_winter), !is.na(Fluke)), aes(x = Fluke, y = PD, color = Fluke, fill = Fluke))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(labels = c("Absent (0)", "Present (1)"),values=c("#98A092","#F14747"))+
  scale_fill_manual(labels = c("Absent (0)", "Present (1)"),values=c("#98A092","#F14747"))+
  geom_point(aes(col = Fluke))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Faith's PD")+
  xlab("")+ theme(legend.position = "none")

alpha.fluke <- ggarrange(P1,P2,P3, ncol = 3) + 
  theme_classic()
```

```{r}
str(sample_data(deer_winter)$Age_Estimate)

P1 <- ggplot(sample_data(deer_winter), aes(x = Age_Estimate, y = Observed, color = Age_Estimate, fill = Age_Estimate))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(name = "Age", labels = c("A", "SA", "C"), values=c("#A55C02","#EA8100","#E3BA88"))+
  scale_fill_manual(name = "Age", labels = c("A", "SA", "C"), values=c("#A55C02","#EA8100","#E3BA88"))+
  geom_point(aes(col = Age_Estimate))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Number of observed ASVs")+
  xlab("")

P2 <- ggplot(sample_data(deer_winter), aes(x = Age_Estimate, y = Shannon, color = Age_Estimate, fill = Age_Estimate))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(name = "Age", labels = c("A", "SA", "C"), values=c("#A55C02","#EA8100","#E3BA88"))+
  scale_fill_manual(name = "Age", labels = c("A", "SA", "C"), values=c("#A55C02","#EA8100","#E3BA88"))+
  geom_point(aes(col = Age_Estimate))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Shannon Index")+
  xlab("")

P3 <- ggplot(sample_data(deer_winter), aes(x = Age_Estimate, y = PD, color = Age_Estimate, fill = Age_Estimate))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(name = "Age", labels = c("A", "SA", "C"), values=c("#A55C02","#EA8100","#E3BA88"))+
  scale_fill_manual(name = "Age", labels = c("A", "SA", "C"), values=c("#A55C02","#EA8100","#E3BA88"))+
  geom_point(aes(col = Age_Estimate))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Faith's PD")+
  xlab("")

alpha.age <- ggarrange(P1,P2,P3, ncol = 3, common.legend = TRUE, legend = "right") + 
  theme_classic()
```

```{r}
str(sample_data(deer_winter)$Sex)

P1 <- ggplot(data=subset(sample_data(deer_winter), !is.na(Sex)), aes(x = Sex, y = Observed, color = Sex, fill = Sex))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  scale_fill_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  geom_point(aes(col = Sex))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Number of observed ASVs")+
  xlab("")

P2 <- ggplot(data=subset(sample_data(deer_winter), !is.na(Sex)), aes(x = Sex, y = Shannon, color = Sex, fill = Sex))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  scale_fill_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  geom_point(aes(col = Sex))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Shannon Index")+
  xlab("")

P3 <- ggplot(data=subset(sample_data(deer_winter), !is.na(Sex)), aes(x = Sex, y = PD, color = Sex, fill = Sex))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  scale_fill_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  geom_point(aes(col = Sex))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Faith's PD")+
  xlab("")

alpha.sex <- ggarrange(P1,P2,P3, ncol = 3, common.legend = TRUE, legend = "right") + 
  theme_classic()
```

```{r}
sample_data(deer_winter)$Site_2 <- as.factor(sample_data(deer_winter)$Site_2)
str(sample_data(deer_winter)$Site_2)

P1 <- ggplot(sample_data(deer_winter), aes(x = Site_2, y = Observed, color = Site_2, fill = Site_2))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = Site_2))+
  scale_color_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  scale_fill_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Number of ASVs")+
  xlab("")+ theme(legend.position = "none")

P2 <- ggplot(sample_data(deer_winter), aes(x = Site_2, y = Shannon, color = Site_2, fill = Site_2))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = Site_2))+
  scale_color_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  scale_fill_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Shannon Index")+
  xlab("")+ theme(legend.position = "none")

P3 <- ggplot(sample_data(deer_winter), aes(x = Site_2, y = PD, color = Site_2, fill = Site_2))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = Site_2))+
  scale_color_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  scale_fill_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Faith's PD")+
  xlab("")+ theme(legend.position = "none")

alpha.sites <- ggarrange(P1,P2,P3, ncol = 3) + 
  theme_classic()
```

```{r}
str(sample_data(deer_winter)$HuntingMonth)

# c("#004aad","#0d00a4", "#22007c", "#140152", "#04052e", "#02010a") # different color vector

P1 <- ggplot(sample_data(deer_winter), aes(x = HuntingMonth, y = Observed, color = HuntingMonth, fill = HuntingMonth))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = HuntingMonth))+
  scale_color_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  scale_fill_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Number of observed ASVs")+
  xlab("")

P2 <- ggplot(sample_data(deer_winter), aes(x = HuntingMonth, y = Shannon, color = HuntingMonth, fill = HuntingMonth))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = HuntingMonth))+
    scale_color_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  scale_fill_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Shannon Index")+
  xlab("")

P3 <- ggplot(sample_data(deer_winter), aes(x = HuntingMonth, y = PD, color = HuntingMonth, fill = HuntingMonth))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = HuntingMonth))+
    scale_color_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  scale_fill_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Faith's PD")+
  xlab("")

alpha.months <- ggarrange(P1,P2,P3, ncol = 3, common.legend = TRUE, legend = "right") + 
  theme_classic()
```

```{r}
# low = "#EEACB7", high = "#760114"  # different color gradient

p1 <- ggplot(sample_data(deer_winter), aes(x = as.numeric(F_Count), y = Observed, fill = F_Count))+
  geom_point(aes(fill =F_Count), pch = 21, size = 2)+
  theme_bw(base_size=14)+
  scale_fill_gradient(low = "#FFF0F3",high = "#F14747",space = "Lab",na.value = "grey50",guide = "colourbar",aesthetics = "fill")+
  ylab("Number of ASVs")+ 
  xlab("")+ theme(legend.position = "none")

p2 <- ggplot(sample_data(deer_winter), aes(x = as.numeric(F_Count), y = Shannon, fill = F_Count))+
  geom_point(aes(fill =F_Count), pch = 21, size = 2)+
  theme_bw(base_size=14)+
  scale_fill_gradient(low = "#FFF0F3",high = "#F14747",space = "Lab",na.value = "grey50",guide = "colourbar",aesthetics = "fill")+
  ylab("Shannon Index")+ 
  xlab("")+ theme(legend.position = "none")

p3 <- ggplot(sample_data(deer_winter), aes(x = as.numeric(F_Count), y = PD, fill = F_Count))+
  geom_point(aes(fill =F_Count), pch = 21, size = 2)+
  theme_bw(base_size=14)+
  scale_fill_gradient(low = "#FFF0F3",high = "#F14747",space = "Lab",na.value = "grey50",guide = "colourbar",aesthetics = "fill")+
  ylab("Faith's PD") + 
  xlab("")+ theme(legend.position = "none")
```

```{r}
alpha.div.fcount.all <- ggarrange(p1,p2,p3, ncol = 3) + 
  theme_classic(base_size=14)
```

```{r}
ggarrange(alpha.fluke, alpha.div.fcount.all, alpha.sites, nrow=1)
```

# Alpha diversity stats

Remove all NAs

```{r}
df_winter <- data.frame(sample_data(deer_winter))
df_winter <- subset(df_winter, Sex != "NA")
df_winter <- subset(df_winter, Fluke != "NA")
df_winter <- subset(df_winter, Age_Estimate != "NA")
df_winter <- subset(df_winter, HuntingMonth != "NA")
```

Fluke Prevalence

```{r}
# original levels
df_winter$Age_Estimate <- factor(df_winter$Age_Estimate, levels = c("Subadult","Calf","Adult"))
df_winter$Site_2 <- factor(df_winter$Site_2, levels = c("Bavarian_Forest","Sumava","Neureichenau"))
df_winter$HuntingMonth <- factor(df_winter$HuntingMonth, levels = c("January","August","September","October","November","December"))

m <- lm(log10(Observed) ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- lm(Shannon ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- glm(PD ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter, family = Gamma("inverse"))
summary(m)
plot(m)
plot(allEffects(m))


# relevel
df_winter$Age_Estimate <- factor(df_winter$Age_Estimate, levels = c("Calf","Adult","Subadult"))
df_winter$Site_2 <- factor(df_winter$Site_2, levels = c("Sumava","Bavarian_Forest","Neureichenau"))
df_winter$HuntingMonth <- factor(df_winter$HuntingMonth, levels = c("August","September","October","November","December","January"))

m <- lm(log10(Observed) ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- lm(Shannon ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- glm(PD ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter, family = Gamma("inverse"))
summary(m)
plot(m)
plot(allEffects(m))


df_winter$HuntingMonth <- factor(df_winter$HuntingMonth, levels = c("September","October","November","December","January","August"))

m <- lm(log10(Observed) ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- lm(Shannon ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- glm(PD ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter, family = Gamma("inverse"))
summary(m)
plot(m)
plot(allEffects(m))


df_winter$HuntingMonth <- factor(df_winter$HuntingMonth, levels = c("October","November","December","January","August","September"))

m <- lm(log10(Observed) ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- lm(Shannon ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- glm(PD ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter, family = Gamma("inverse"))
summary(m)
plot(m)
plot(allEffects(m))


df_winter$HuntingMonth <- factor(df_winter$HuntingMonth, levels = c("November","December","January","August","September","October"))

m <- lm(log10(Observed) ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- lm(Shannon ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- glm(PD ~ Sex + Site_2 + Age_Estimate*Fluke + HuntingMonth + Seq_depth, data = df_winter, family = Gamma("inverse"))
summary(m)
plot(m)
plot(allEffects(m))
```

Fluke infection intensity

```{r}
# original levels
df_winter$Age_Estimate <- factor(df_winter$Age_Estimate, levels = c("Subadult","Calf","Adult"))
df_winter$Site_2 <- factor(df_winter$Site_2, levels = c("Bavarian_Forest","Sumava","Neureichenau"))
df_winter$HuntingMonth <- factor(df_winter$HuntingMonth, levels = c("January","August","September","October","November","December"))

m <- lm(log10(Observed) ~ Sex + Site_2 + Age_Estimate + F_Count + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- lm(Shannon ~ Sex + Site_2 + Age_Estimate + F_Count + HuntingMonth + Seq_depth, data = df_winter)
summary(m)
plot(m)
plot(allEffects(m))

m <- glm(PD ~ Sex + Site_2 + Age_Estimate + F_Count + HuntingMonth + Seq_depth, data = df_winter, family = Gamma("inverse"))
summary(m)
plot(m)
plot(allEffects(m))
```

# Beta diversity

Remove the months February - July

```{r}
deer_rare_all <- readRDS("all.deer.rare.RDS")

# convert to relative abundances
deer_rare_all <- microbiome::transform(deer_rare_all, "compositional")


sample_data(deer_rare_all)$F_Count <- as.numeric(sample_data(deer_rare_all)$F_Count)
sample_data(deer_rare_all)$Fluke <- as.factor(sample_data(deer_rare_all)$Fluke)
sample_data(deer_rare_all)$HuntingMonth <- as.factor(sample_data(deer_rare_all)$HuntingMonth)

levels(sample_data(deer_rare_all)$HuntingMonth)
levels(sample_data(deer_rare_all)$HuntingMonth) <- c("April", "August", "December", "February", "January", "July", "June", "June", "March", "May", "November" ,"October", "September") 

sample_data(deer_rare_all)$HuntingMonth <- factor(sample_data(deer_rare_all)$HuntingMonth, levels = c("January", "February", "March", "April",  "May","June","July","August", "September","October","November" ,"December"))

deer_rare_all1 <- subset_samples(deer_rare_all, HuntingMonth != "February")
deer_rare_all1 <- subset_samples(deer_rare_all1, HuntingMonth != "March")
deer_rare_all1 <- subset_samples(deer_rare_all1, HuntingMonth != "April")
deer_rare_all1 <- subset_samples(deer_rare_all1, HuntingMonth != "May")
deer_rare_all1 <- subset_samples(deer_rare_all1, HuntingMonth != "June")
deer_rare_all1 <- subset_samples(deer_rare_all1, HuntingMonth != "July")

table(sample_data(deer_rare_all1)$HuntingMonth)

deer_rare_all <- deer_rare_all1

table(sample_data(deer_rare_all)$HuntingMonth)


summary(glm(Fluke ~ HuntingMonth, data = data.frame(sample_data(deer_rare_all)), family = binomial))

sample_data(deer_rare_all)$HuntingMonth <- factor(sample_data(deer_rare_all)$HuntingMonth, levels = c("August","September","October","November","December","January"))

summary(glm(Fluke ~ HuntingMonth, data = data.frame(sample_data(deer_rare_all)), family = binomial))

sample_data(deer_rare_all)$HuntingMonth <- factor(sample_data(deer_rare_all)$HuntingMonth, levels = c("September","October","November","December","January","August"))

summary(glm(Fluke ~ HuntingMonth, data = data.frame(sample_data(deer_rare_all)), family = binomial))

sample_data(deer_rare_all)$HuntingMonth <- factor(sample_data(deer_rare_all)$HuntingMonth, levels = c("October","November","December","January","August","September"))

summary(glm(Fluke ~ HuntingMonth, data = data.frame(sample_data(deer_rare_all)), family = binomial))

sample_data(deer_rare_all)$HuntingMonth <- factor(sample_data(deer_rare_all)$HuntingMonth, levels = c("November","December","January","August","September","October"))

summary(glm(Fluke ~ HuntingMonth, data = data.frame(sample_data(deer_rare_all)), family = binomial))

sample_data(deer_rare_all)$HuntingMonth <- factor(sample_data(deer_rare_all)$HuntingMonth, levels = c("January","August","September","October","November","December"))
```

No difference in fluke prevalence across hunting months

```{r}
deer_rare_red <- subset_samples(deer_rare_all, Sex != "NA") # remove NAs, because they create an error in the model
deer_rare_red <- subset_samples(deer_rare_red, Age_Estimate != "NA") # remove NAs, because they create an error in the model
deer_rare_red <- subset_samples(deer_rare_red, Fluke != "NA") # remove NAs, because they create an error in the model
deer_rare_red <- subset_samples(deer_rare_red, Site_2 != "NA") # remove NAs, because they create an error in the model
deer_rare_red <- subset_samples(deer_rare_red, HuntingMonth != "NA") # remove NAs, because they create an error in the model

deer_rare_red = prune_taxa(taxa_sums(deer_rare_red) > 0, deer_rare_red) 
deer_rare_red

#saveRDS(deer_rare_red, "deer_rare_red_rel.RDS")

# N=417
```

```{r}
deer_rare_red_rel <- readRDS("deer_rare_red_rel.RDS")
```

```{r}
table(sample_data(deer_rare_red)$Fluke)
```

# Beta Diversity Visualizations and Permutation tests (= difference in composition centroids between groups?)

Visualize beta diversity in constrained ordinations (CAP)

1) Unweighted UniFrac = presence/absence and phylogeny of ASVs

2) Weighted UniFrac = abundance and phylogeny of ASVs

```{r}
deer_rare_red <- readRDS("deer_rare_red_rel.RDS")

sample_data(deer_rare_red)$F_Count

# agglomorate taxa at genus level

deer_rare_red_compositional <- microbiome::transform(deer_rare_red, "compositional")
deer_rare_merged <- tax_glom(deer_rare_red_compositional, taxrank="Genus") # agglomerate taxa to Genus level
# deer_rare_merged
# otu_table()   OTU Table:         [ 370 taxa and 91 samples ]
```

```{r}
otutable<-data.frame(t(deer_rare_merged@otu_table@.Data))
metadata <- data.frame(sample_data(deer_rare_merged))

metadata$F_Count
str(metadata$F_Count)

Sex<-as.factor(metadata$Sex)
Age<-as.factor(metadata$Age_Estimate)
Fluke<-as.factor(metadata$Fluke)
Site_2<-as.factor(metadata$Site_2)
HuntingMonth<-as.factor(metadata$HuntingMonth)
F_Count<-as.numeric(as.character(metadata$F_Count))
Seq_depth<-as.numeric(as.character(metadata$Seq_depth))
```

## Unweighted UniFrac

```{r}
uunifrac_dist<-distance(deer_rare_merged, method = "unifrac")
```

1) Fluke prevalence model

```{r}
fluke_model_uu<-capscale(uunifrac_dist ~ Sex +
                           Site_2 + 
                           Age*Fluke + 
                           HuntingMonth,
                         env = metadata,
                         comm = otutable)
```

```{r}
set.seed(1) # set seed for reproducibility
anova_uunifrac2<-anova.cca(fluke_model_uu, by="terms", permutations = 9999)
anova_uunifrac2
```

Extract model data

```{r}
fluke_model_uu_df<-scores(fluke_model_uu)
```

Extract CAP scores

```{r}
vectors_df<-data.frame(fluke_model_uu_df$sites)
vectors_df$Feature.id<-row.names(vectors_df)
```

```{r}
sample_data(deer_rare_merged)$Fluke
sample_data(deer_rare_merged)$Feature.id

vectors_df$Feature.id == sample_data(deer_rare_merged)$Feature.id

sample_metadata<-data.frame(sample_data(deer_rare_merged))[,c("Feature.id", "Fluke")]
vectors_df<-merge(vectors_df, metadata, by = "Feature.id")
```

```{r}
centroids_df<-data.frame(fluke_model_uu_df$centroids)
centroids_df<-centroids_df[1:16,]
centroids_df
row.names(centroids_df)<-c("Sex:F","Sex:M","BF-NP","N-FE","S-NP","Age:A","Age:SA","Age:C","Fluke:0","Fluke:1","Jan","Aug","Sep","Oct","Nov","Dec")
```

```{r}
vectors_uunifrac<-vectors_df
centroids_uunifrac<-centroids_df
```

Fluke

```{r}
ggplot(vectors_uunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Fluke, col = Fluke), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Fluke), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw() + 
  scale_fill_manual(values=c("#98A092","#F14747"))+
  scale_color_manual(values=c("#98A092","#F14747"))+
  
  # add arrows
  
  geom_segment(data=centroids_uunifrac[9:10,], aes(x = 0, y = 0, xend = CAP1*5, yend = CAP2*5),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_uunifrac[9:10,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_uunifrac[9:10,])))+
  
  theme_light(base_size = 14)
```

Sex

```{r}
ggplot(vectors_uunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Sex, col = Sex), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Sex), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#F6DE22","#60712F"))+
  scale_color_manual(values=c("#F6DE22","#60712F"))+
  
  # add arrows
  
  geom_segment(data=centroids_uunifrac[1:2,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_uunifrac[1:2,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_uunifrac[1:2,])))+
  
  theme_light(base_size = 14)
```

Age

```{r}
ggplot(vectors_uunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Age, col = Age), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Age), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(name = "Age", labels = c("A", "SA", "C"), values=c("#A55C02","#EA8100","#E3BA88"))+
  scale_color_manual(name = "Age", labels = c("A", "SA", "C"), values=c("#A55C02","#EA8100","#E3BA88"))+
  
  # add arrows
  
  geom_segment(data=centroids_uunifrac[6:8,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_uunifrac[6:8,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_uunifrac[6:8,])))+
  
  theme_light(base_size = 14)
```

Sites

```{r}
ggplot(vectors_uunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Site_2, col = Site_2), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Site_2), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#3D77A4","#A3CBE1","#A2E086"))+
  scale_color_manual(values=c("#3D77A4","#A3CBE1","#A2E086"))+
  
  # add arrows
  
  geom_segment(data=centroids_uunifrac[3:5,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_uunifrac[3:5,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_uunifrac[3:5,])))+
  theme_light(base_size = 14) + theme(legend.position = "none")
```

Month

```{r}
ggplot(vectors_uunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = HuntingMonth, col = HuntingMonth), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =HuntingMonth), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  scale_color_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  
  # add arrows
  
  geom_segment(data=centroids_uunifrac[11:16,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_uunifrac[11:16,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_uunifrac[11:16,])))+
  
  theme_light(base_size = 14)
```


2) Fluke intensity model

```{r}
fcount_model_uu <-capscale(uunifrac_dist ~
                             Sex + 
                             Site_2 + 
                             Age + 
                             F_Count + 
                             HuntingMonth,
                           env = metadata,
                           comm = otutable)
```

```{r}
fcount_model_uu
summary(fcount_model_uu)$concont
plot(fcount_model_uu)
```

```{r}
set.seed(1) # set seed for reproducibility
anova_uunifrac<-anova.cca(fcount_model_uu, by="terms", permutations = 9999)
anova_uunifrac
```

Extract model data

```{r}
fcount_model_uu_df<-scores(fcount_model_uu)
```

Get appropriate arrow multiplier

```{r}
scrs<-scores(fcount_model_uu, display="bp")
scrs

mul<-vegan:::ordiArrowMul(scrs)
mul
# 2.619505

scr_mul <- scrs * mul
scr_mul <- data.frame(scr_mul)
```

Extract CAP scores

```{r}
vectors_df<-data.frame(fcount_model_uu_df$sites)
vectors_df$Feature.id<-row.names(vectors_df)
```

```{r}
sample_data(deer_rare_merged)$F_Count
sample_data(deer_rare_merged)$Feature.id

vectors_df$Feature.id == sample_data(deer_rare_merged)$Feature.id    # double-check there the same

sample_metadata<-data.frame(sample_data(deer_rare_merged))[,c("Feature.id", "F_Count")]
vectors_df<-merge(vectors_df, metadata, by = "Feature.id")
```

Extract group centroids and rename variables 

```{r}
centroids_df<-data.frame(fcount_model_uu_df$centroids)
centroids_df<-centroids_df[1:14,]
centroids_df
row.names(centroids_df)<-c("Sex:F","Sex:M","BF-NP","N-FE","S-NP","Age:A","Age:SA","Age:C","Jan","Aug","Sep","Oct","Nov","Dec")
```

```{r}
vectors_unifrac<-vectors_df
centroids_unifrac<-centroids_df
```

F_Count

```{r}
ggplot(vectors_unifrac, aes(x = CAP1, y = CAP2))+
  
  geom_point(aes(fill =F_Count), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  theme_bw() + 
  scale_fill_gradient(low = "#FFF0F3",high = "#F14747",space = "Lab",na.value = "grey50",guide = "colourbar",aesthetics = "fill")+
  
  # add arrows
  
  geom_segment(data=scr_mul[6,], aes(x = 0, y = 0, xend = CAP1*5, yend = CAP2*4),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=scr_mul[6,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(scr_mul[6,])))+
  
  theme_light(base_size = 14)
```

Age

```{r}
ggplot(vectors_unifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Age, col = Age), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Age), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#A55C02","#EA8100","#E3BA88"))+
  scale_color_manual(values=c("#A55C02","#EA8100","#E3BA88"))+
  
  # add arrows
  
  geom_segment(data=centroids_unifrac[6:8,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_unifrac[6:8,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_unifrac[6:8,])))+
  
  theme_light(base_size = 14)
```

Sex

```{r}
ggplot(vectors_unifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Sex, col = Sex), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Sex), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#F6DE22","#60712F"))+
  scale_color_manual(values=c("#F6DE22","#60712F"))+
  
  # add arrows
  
  geom_segment(data=centroids_unifrac[1:2,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_unifrac[1:2,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_unifrac[1:2,])))+
  
  theme_light(base_size = 14)
```

Sites

```{r}
ggplot(vectors_unifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Site_2, col = Site_2), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Site_2), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#3D77A4","#A3CBE1","#A2E086"))+
  scale_color_manual(values=c("#3D77A4","#A3CBE1","#A2E086"))+
  
  # add arrows
  
  geom_segment(data=centroids_unifrac[3:5,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_unifrac[3:5,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_unifrac[3:5,])))+
  
  theme_light(base_size = 14)
```

Month

```{r}
ggplot(vectors_unifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = HuntingMonth, col = HuntingMonth), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =HuntingMonth), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  scale_color_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  
  # add arrows
  
  geom_segment(data=centroids_unifrac[9:14,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_unifrac[9:14,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_unifrac[9:14,])))+
  
  theme_light(base_size = 14)
```

## Weighted UniFrac

```{r}
wunifrac_dist<-distance(deer_rare_merged, method = "wunifrac")
```

1) Fluke prevalence model

```{r}
fluke_model<-capscale(wunifrac_dist ~ Sex +
                        Site_2 + 
                        Age*Fluke + 
                        HuntingMonth,
                      env = metadata,
                      comm = otutable)
```

```{r}
set.seed(1) # set seed for reproducibility
anova_wunifrac2<-anova.cca(fluke_model, by="terms", permutations = 9999)
anova_wunifrac2
```

Extract data from model

```{r}
fluke_model_df<-scores(fluke_model)
```

Extract CAP scores

```{r}
vectors_df<-data.frame(fluke_model_df$sites)
vectors_df$Feature.id<-row.names(vectors_df)
```

```{r}
sample_data(deer_rare_merged)$Fluke
sample_data(deer_rare_merged)$Feature.id

vectors_df$Feature.id == sample_data(deer_rare_merged)$Feature.id

sample_metadata<-data.frame(sample_data(deer_rare_merged))[,c("Feature.id", "Fluke")]
vectors_df<-merge(vectors_df, metadata, by = "Feature.id")
```

```{r}
centroids_df<-data.frame(fluke_model_df$centroids)
centroids_df<-centroids_df[1:16,]
centroids_df
row.names(centroids_df)<-c("Sex:F","Sex:M","BF-NP","N-FE","S-NP","Age:A","Age:SA","Age:C","Fluke:0","Fluke:1","Jan","Aug","Sep","Oct","Nov","Dec")
```

```{r}
vectors_wunifrac<-vectors_df
centroids_wunifrac<-centroids_df
```

Fluke

```{r}
ggplot(vectors_wunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Fluke, col = Fluke), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Fluke), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw() + 
  scale_fill_manual(values=c("#98A092","#F14747"))+
  scale_color_manual(values=c("#98A092","#F14747"))+
  
  # add arrows
  
  geom_segment(data=centroids_wunifrac[9:10,], aes(x = 0, y = 0, xend = CAP1*5, yend = CAP2*5),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_wunifrac[9:10,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_wunifrac[9:10,])))+
  
  theme_light(base_size = 14)
```

Sex

```{r}
ggplot(vectors_wunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Sex, col = Sex), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Sex), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#F6DE22","#60712F"))+
  scale_color_manual(values=c("#F6DE22","#60712F"))+
  
  # add arrows
  
  geom_segment(data=centroids_wunifrac[1:2,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_wunifrac[1:2,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_wunifrac[1:2,])))+
  
  theme_light(base_size = 14)
```

Age

```{r}
ggplot(vectors_wunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Age, col = Age), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Age), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#A55C02","#EA8100","#E3BA88"))+
  scale_color_manual(values=c("#A55C02","#EA8100","#E3BA88"))+
  
  # add arrows
  
  geom_segment(data=centroids_wunifrac[6:8,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_wunifrac[6:8,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_wunifrac[6:8,])))+
  
  theme_light(base_size = 14)
```

Sites

```{r}
ggplot(vectors_wunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Site_2, col = Site_2), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Site_2), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_color_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  scale_fill_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  
  # add arrows
  
  geom_segment(data=centroids_wunifrac[3:5,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_wunifrac[3:5,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_wunifrac[3:5,])))+
  theme_light(base_size = 14)
```

Month

```{r}
ggplot(vectors_wunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = HuntingMonth, col = HuntingMonth), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =HuntingMonth), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  scale_color_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  
  # add arrows
  
  geom_segment(data=centroids_wunifrac[11:16,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_wunifrac[11:16,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_wunifrac[11:16,])))+
  
  theme_light(base_size = 14)
```

2) Fluke intensity model

```{r}
fcount_model<-capscale(wunifrac_dist ~
                         Sex + 
                         Site_2 + 
                         Age + 
                         F_Count + 
                         HuntingMonth,
                       env = metadata,
                       comm = otutable)
```

```{r}
fcount_model
summary(fcount_model)$concont
plot(fcount_model)
```

```{r}
set.seed(1) # set seed for reproducibility
anova_wunifrac<-anova.cca(fcount_model, by="terms", permutations = 9999)
anova_wunifrac
```

Extract model data

```{r}
fcount_model_df<-scores(fcount_model)
```

Get appropriate arrow multiplier

```{r}
scrs<-scores(fcount_model, display="bp")
scrs

mul<-vegan:::ordiArrowMul(scrs)
mul
#  2.181022

scr_mul <- scrs * mul
scr_mul <- data.frame(scr_mul)
```

Extract CAP scores

```{r}
vectors_df<-data.frame(fcount_model_df$sites)
vectors_df$Feature.id<-row.names(vectors_df)
```

```{r}
sample_data(deer_rare_merged)$F_Count
sample_data(deer_rare_merged)$Feature.id

vectors_df$Feature.id == sample_data(deer_rare_merged)$Feature.id

sample_metadata<-data.frame(sample_data(deer_rare_merged))[,c("Feature.id", "F_Count")]
vectors_df<-merge(vectors_df, metadata, by = "Feature.id")
```

```{r}
centroids_df<-data.frame(fcount_model_df$centroids)
centroids_df<-centroids_df[1:14,]
centroids_df
row.names(centroids_df)<-c("Sex:F","Sex:M","BF-NP","N-FE","S-NP","Age:A","Age:SA","Age:C","Jan","Aug","Sep","Oct","Nov","Dec")
```

```{r}
vectors_wunifrac<-vectors_df
centroids_wunifrac<-centroids_df
species_wunifrac<-species_scores
```

F_Count

```{r}
ggplot(vectors_wunifrac, aes(x = CAP1, y = CAP2))+
  
  geom_point(aes(fill =F_Count), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  theme_bw() + 
  scale_fill_gradient(low = "#FFF0F3",high = "#F14747",space = "Lab",na.value = "grey50",guide = "colourbar",aesthetics = "fill")+
  
  # add arrows
  
  geom_segment(data=scr_mul[6,], aes(x = 0, y = 0, xend = CAP1*5, yend = CAP2*5),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=scr_mul[6,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(scr_mul[6,])))+
  
  theme_light(base_size = 14)
```

Age

```{r}
ggplot(vectors_wunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Age, col = Age), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Age), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#A55C02","#EA8100","#E3BA88"))+
  scale_color_manual(values=c("#A55C02","#EA8100","#E3BA88"))+
  
  # add arrows
  
  geom_segment(data=centroids_wunifrac[6:8,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_wunifrac[6:8,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_wunifrac[6:8,])))+
  
  theme_light(base_size = 14)
```

Sites

```{r}
ggplot(vectors_wunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = Site_2, col = Site_2), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =Site_2), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#3D77A4","#A3CBE1","#A2E086"))+
  scale_color_manual(values=c("#3D77A4","#A3CBE1","#A2E086"))+
  
  # add arrows
  
  geom_segment(data=centroids_wunifrac[3:5,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_wunifrac[3:5,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_wunifrac[3:5,])))+
  
  theme_light(base_size = 14)
```

Month

```{r}
ggplot(vectors_wunifrac, aes(x = CAP1, y = CAP2))+
  
  stat_ellipse(geom = "polygon", aes(fill = HuntingMonth, col = HuntingMonth), level = 0.9, alpha = 0.3, size = 0.5)+
  geom_point(aes(fill =HuntingMonth), pch = 21, size = 3, alpha = 1, stroke = 1, col = "black")+
  
  theme_bw()+
  scale_fill_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  scale_color_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  
  # add arrows
  
  geom_segment(data=centroids_wunifrac[9:14,], aes(x = 0, y = 0, xend = CAP1*2, yend = CAP2*2),
               arrow = arrow(length = unit(0.5, "cm"), type = "closed"), lwd = 1,  col = "black")+
  ggrepel::geom_label_repel(data=centroids_wunifrac[9:14,],  
                            alpha = 0.9, col = "black", size = 4, fill = "white", 
                            aes(CAP1*2, CAP2*2, label = row.names(centroids_wunifrac[9:14,])))+
  
  theme_light(base_size = 14)
```

## Differential abundance

1) Fluke infection

Genus level

```{r}
#deer_winter_inf_comp <- readRDS("deer_winter.RDS")

# remove individuals without fluke info
deer_winter_inf_comp <- subset_samples(deer_winter_inf_comp, Fluke != "NA")

# remove taxa with zero counts
deer_winter_inf_comp = prune_taxa(taxa_sums(deer_winter_inf_comp) > 0, deer_winter_inf_comp) 

# agglomerate to Genus level
deer_winter_inf_comp <- tax_glom(deer_winter_inf_comp, taxrank = "Genus")

# extract data frame of bacterial genera for taxonomic assigment later
deer_winter_inf_comp_df <- data.frame(tax_table(deer_winter_inf_comp)[,"Genus"])
```

ALDEx

```{r}
library(ALDEx2)

# Generate Monte Carlo samples of the Dirichlet distribution for each sample.
# Convert each instance using the centred log-ratio transform.
# This is the input for all further analyses.

x_fluke_infection <- aldex.clr(
  reads = otu_table(deer_winter_inf_comp),
  conds = sample_data(deer_winter_inf_comp)$Fluke, 
  # 128 recommened for ttest, 1000 for rigorous effect size calculation
  mc.samples = 128, 
  denom = "all",
  verbose = FALSE)


# calculates expected values of the Welch's t-test and Wilcoxon rank test on
# the data returned by aldex.clr

tt_fluke_infection <- aldex.ttest(x_fluke_infection, paired.test = FALSE, verbose = FALSE)

# determines the median clr abundance of the feature in all samples and in
# groups, the median difference between the two groups, the median variation
# within each group and the effect size, which is the median of the ratio
# of the between group difference and the larger of the variance within groups

effect_fluke_infection <- aldex.effect(x_fluke_infection, CI = TRUE, verbose = FALSE)


# combine all outputs 
aldex_out_fluke_infection <- data.frame(tt_fluke_infection, effect_fluke_infection)

rownames(aldex_out_fluke_infection)
rownames(deer_winter_inf_comp_df) 

df_merge <- merge(aldex_out_fluke_infection, deer_winter_inf_comp_df,
                  by = 'row.names', all = TRUE)


aldex_out_fluke_infection <- df_merge

View(aldex_out_fluke_infection)

##### plot whichever ####

par(mfrow = c(1, 2))
aldex.plot(
  aldex_out_fluke_infection, 
  type = "MA", 
  test = "welch", 
  xlab = "Log-ratio abundance",
  ylab = "Difference",
  cutoff = 0.05
)

aldex.plot(
  aldex_out_fluke_infection, 
  type = "MW", 
  test = "welch",
  xlab = "Dispersion",
  ylab = "Difference",
  cutoff = 0.05
)

# read out results
sig.subset<-rownames_to_column(aldex_out_fluke_infection, "genus") %>%
  filter(wi.eBH <= 0.05) # Wilcoxon Rank Sum test p-value for each taxon, Benjamini-Hochberg corrected 


aldex_out_fluke_infection

left<-subset(aldex_out_fluke_infection, effect>0.0)
right<-subset(aldex_out_fluke_infection, effect<0.0)
```

2) Study sites

BF-NP vs. S-NP (for the other site comparisons the sample sizes are biased and thus will not be analyzed)

Subset only the relevant sites

```{r}
deer_bf_su <- subset_samples(deer_winter, Site_2 != "Neureichenau")
sample_data(deer_bf_su)$Site_2
deer_bf_su = prune_taxa(taxa_sums(deer_bf_su) > 0, deer_bf_su) 
```

Agglomerate taxa on Genus level

```{r}
deer_bf_su <- tax_glom(deer_bf_su, taxrank = "Genus")
#saveRDS(deer_bf_su, "deer_bf_su_genus.RDS")
deer_bf_su_df <- data.frame(tax_table(deer_bf_su)[,"Genus"])
```

Aldex

```{r}
# Generate Monte Carlo samples of the Dirichlet distribution for each sample.
# Convert each instance using the centred log-ratio transform.
# This is the input for all further analyses.

x_bf_su <- aldex.clr(
  reads = otu_table(deer_bf_su),
  conds = sample_data(deer_bf_su)$Site_2, 
  # 128 recommened for ttest, 1000 for rigorous effect size calculation
  mc.samples = 128, 
  denom = "all",
  verbose = FALSE)


# calculates expected values of the Welch's t-test and Wilcoxon rank test on
# the data returned by aldex.clr

tt_bf_su <- aldex.ttest(x_bf_su, paired.test = FALSE, verbose = FALSE)


# determines the median clr abundance of the feature in all samples and in
# groups, the median difference between the two groups, the median variation
# within each group and the effect size, which is the median of the ratio
# of the between group difference and the larger of the variance within groups

effect_bf_su <- aldex.effect(x_bf_su, CI = TRUE, verbose = FALSE)


# combine all outputs 
aldex_out_bf_su <- data.frame(tt_bf_su, effect_bf_su)

rownames(aldex_out_bf_su)
rownames(deer_bf_su_df) 

df_merge <- merge(aldex_out_bf_su, deer_bf_su_df,
                  by = 'row.names', all = TRUE)


aldex_out_bf_su <- df_merge

View(aldex_out_bf_su)

##### plot whichever ####

par(mfrow = c(1, 2))
aldex.plot(
  aldex_out_bf_su, 
  type = "MA", 
  test = "welch", 
  xlab = "Log-ratio abundance",
  ylab = "Difference",
  cutoff = 0.05
)

aldex.plot(
  aldex_out_bf_su, 
  type = "MW", 
  test = "welch",
  xlab = "Dispersion",
  ylab = "Difference",
  cutoff = 0.05
)

# read out results

aldex_out_bf_su

#write.csv(aldex_out_bf_su, "aldex_out_bf_su.csv")

sig.subset<-rownames_to_column(aldex_out_bf_su, "genus") %>%
  filter(wi.eBH <= 0.05) # Wilcoxon Rank Sum test p-value for each taxon, Benjamini-Hochberg corrected 

sig.subset.bf.su <- sig.subset
#write.csv(sig.subset.bf.su, "sig.subset.bf.su_new.csv")

left<-subset(sig.subset, effect>0.0)
right<-subset(sig.subset, effect<0.0)


plot(aldex_out_bf_su$effect, aldex_out_bf_su$wi.eBH, log="y", cex=0.7,
     pch=19, xlab="Effect size", ylab="P value", main="Bavarian Forest vs.Sumava")+
  points(right$effect, right$wi.eBH, cex=0.7, col="#4CAF50",pch=19)+
  points(left$effect, left$wi.eBH, cex=0.7, col="#FDD835",pch=19)+
  abline(h=0.05, lty=2, col="grey")+
  text(wi.eBH~effect, labels=Genus, data=left, cex=0.9, font=2, pos=2)+
  text(wi.eBH~effect, labels=Genus, data=right, cex=0.9, font=2, pos=4)
```

Visualize differentially abundant bacterial taxa between BF-NP and S-NP

```{r}
deer_bf_su <- readRDS("deer_bf_su_genus.RDS")
```

```{r}
deer_tax <- data.frame(tax_table(deer_bf_su))

#names <- deer_tax$Genus
#test<- as.matrix(cbind(deer_tax, names))
#tax_table(deer_bf_su) <- test # map back

my_aldex_genera <- subset_taxa(deer_bf_su, Genus=="Akkermansia"|
                                 Genus=="Alistipes"|
                                 Genus=="Anaerosporobacter"|
                                 Genus=="Bacteroides"|
                                 Genus=="Candidatus_Saccharimonas"|
                                 Genus=="Christensenellaceae_R-7_group"|
                                 Genus=="Clostridium_sensu_stricto_1"|
                                 Genus=="dgA-11_gut_group"|
                                 Genus=="Dorea"|
                                 Genus=="Escherichia-Shigella"|
                                 Genus=="[Eubacterium]_siraeum_group"|
                                 Genus=="Flavonifractor"|
                                 Genus=="Gastranaerophilales"|
                                 Genus=="hoa5-07d05_gut_group"|
                                 Genus=="Lachnospiraceae_UCG-001"|
                                 Genus=="Mailhella"|
                                 Genus=="p-251-o5"|
                                 Genus=="Prevotellaceae_UCG-004"|
                                 Genus=="Roseburia"|
                                 Genus=="Sphingomonas"|
                                 Genus=="Turicibacter"|
                                 Genus=="UCG-005"|
                                 Genus=="UCG-009"|
                                 Genus=="UCG-010"|
                                 Genus=="UCG-011"|
                                 Genus=="Variovorax")



my_aldex_genera.df <- data.frame(otu_table(my_aldex_genera))

# add +1 to all values in the dataframe, then we are able to compare abundances later properly with log transformation
my_aldex_genera.df_1 <- my_aldex_genera.df + 1  # first save into new object, double check is correct
my_aldex_genera.df <- my_aldex_genera.df_1      # then rename

my_aldex_genera.df  = cbind(as(my_aldex_genera.df, "data.frame"), as(tax_table(my_aldex_genera)[rownames(my_aldex_genera.df), ], "matrix"))

my_aldex_genera.df = subset(my_aldex_genera.df, select = -c(Kingdom,Phylum, Class, Order, Family, Genus, Species))

my_aldex_genera.df<-data.frame(t(my_aldex_genera.df))
head(my_aldex_genera.df)
tail(my_aldex_genera.df)

names(my_aldex_genera.df)

# check if rows are ordered similar in metadata
metadata<-data.frame(sample_data(my_aldex_genera))
head(metadata)
tail(metadata)

# both metadata and asv-df are in the same order, thus we can bind together

write.csv(my_aldex_genera.df, "my_aldex_genera_df.csv")
write.csv(metadata, "metadata.csv")


metadata1<-cbind(metadata, my_aldex_genera.df)


head(metadata1)
names(metadata1)

head(table(metadata1$Flavonifractor))
head(table(my_aldex_genera.df$X7d11b06e6d425e8724f2e674e4c04b05, my_aldex_genera.df$Feature.id))

# Change colnames of ASV columns to correct and short names
colnames(metadata1)[colnames(metadata1) %in% c("X12a9f31a1adb993fc4aeeea3856b48d8",
                                               "e32131ae35ffb64f9274b169428daae3",
                                               "daa765b8dc0bb192164d5d4ac25a8cc1",
                                               "X1247becb9f4350af3d89029ac6b4f3a1",
                                               "X314b9d7b7bf737a1c4083de527c94322",
                                               "fab326ad858072c48453ebf53086279d",
                                               "X1b9c4e175d6198d5af9dc9380f13290e",
                                               "X7784d226c082c0c71e026f3441f4543f",
                                               "X60c7287edd65280f64913dd35efbbce3",
                                               "e7b8c901dd09babc6664ab6d9487b29f",
                                               "X9279796609d6f137c395fb3d88814a8d",
                                               "X476098ccd78c639f3675a55f3c5ef9f5",
                                               "X349114f810a17493c38a26e1cacfb6dd",
                                               "c913d815f877f24a2b4dc1742f8c49fe",
                                               "X2960b21d5dc2229c63cc7f701e0b705e",
                                               "X52e434ddc810aa11d27e5ea6c98f4d56",
                                               "a9d6c8eb01c5859aa574583bac00d8c7",
                                               "dfd06ecac6b0e634fa2eb3cc5cbedcf8",
                                               "X2c7f8615fa36da18c6213391c598de0",
                                               "e1fe9d81d36f9cbf45777bdf197f1aa9",
                                               "X30fced9f9b540bc567f71546bdb63861",
                                               "b0c9a5f373abf28d8ce11d0f26ad210e",
                                               "b1e46a99a529379724a552f65873f9bd",
                                               "X7da2a65f944b4160f64214b414f49c80",
                                               "X7d11b06e6d425e8724f2e674e4c04b05",
                                               "X2b7a1553d43d19cc025a90528b74cbf6")] <- c("Sphingomonas",
                                                                                          "Akkermansia",
                                                                                          "Turicibacter",
                                                                                          "Prevotellaceae_UCG-004",
                                                                                          "Bacteroides",
                                                                                          "Candidatus_Saccharimonas",
                                                                                          "Christensenellaceae_R-7_group",
                                                                                          "Variovorax",
                                                                                          "hoa5-07d05_gut_group",
                                                                                          "UCG-005",
                                                                                          "Alistipes",
                                                                                          "Dorea",
                                                                                          "Escherichia-Shigella",
                                                                                          "Mailhella",
                                                                                          "Anaerosporobacter",
                                                                                          "UCG-010",
                                                                                          "Clostridium_sensu_stricto_1",
                                                                                          "Lachnospiraceae_UCG-001",
                                                                                          "Eubacterium_siraeum_group",
                                                                                          "Roseburia",
                                                                                          "dgA-11_gut_group",
                                                                                          "UCG-009",
                                                                                          "UCG-011",
                                                                                          "p-251-o5",
                                                                                          "Flavonifractor",
                                                                                          "Gastranaerophilales")




metadata1

#aggregate(metadata1[52:77], list(metadata1$Site_2), FUN=mean)
#aggregate(metadata1[52:77], list(metadata1$Site_2), FUN=median)


#metadata1 %>%
#  group_by(Site_2) %>%
#  summarise_at(vars(Sphingomonas:Dorea), list(name = median))


# calcualte mean abundance per site and add to df

df <- data.frame(metadata1 %>%
  group_by(Site_2) %>%
  summarise_at(vars(Sphingomonas:Dorea), list(mean)))

# make long format

df_long <- gather(df, Genus, MeanAbundance,  Sphingomonas:Dorea, factor_key=TRUE)
df_long$Genus
```

Subset relevant taxa

```{r}
df_long_BNP <- subset(df_long, Genus== "Sphingomonas" |
                        Genus== "Variovorax" |
                        Genus== "Clostridium_sensu_stricto_1" |
                        Genus== "Turicibacter" |
                        Genus== "UCG.010" |
                        Genus== "Eubacterium_siraeum_group" |
                        Genus== "Flavonifractor" |
                        Genus== "UCG.005" |
                        Genus== "Christensenellaceae_R.7_group" |
                        Genus== "UCG.011" |
                        Genus== "Anaerosporobacter" |
                        Genus== "Lachnospiraceae_UCG.001")


df_long_SNP <- subset(df_long, Genus== "Mailhella" |
                        Genus== "UCG.009" |
                        Genus== "Candidatus_Saccharimonas" |
                        Genus== "Gastranaerophilales" |
                        Genus== "hoa5.07d05_gut_group" |
                        Genus== "Bacteroides" |
                        Genus== "Prevotellaceae_UCG.004" |
                        Genus== "p.251.o5" |
                        Genus== "dgA.11_gut_group" |
                        Genus== "Akkermansia" |
                        Genus== "Alistipes" |
                        Genus== "Roseburia" |
                        Genus== "Escherichia.Shigella" |
                        Genus== "Dorea")
```

Create labels

```{r}
bnp_labels<- c("Sphingomonas",
                 "Variovorax",
                 "Clostridium_sensu_stricto_1",
                 "Turicibacter",
                 "UCG.010",
                 "Eubacterium_siraeum_group",
                 "Flavonifractor",
                 "UCG.005",
                 "Christensenellaceae_R.7_group",
                 "UCG.011",
                 "Anaerosporobacter",
                 "Lachnospiraceae_UCG.001")

snp_labels <- c("Mailhella",
                 "UCG.009",
                 "Candidatus_Saccharimonas",
                 "Gastranaerophilales",
                 "hoa5.07d05_gut_group",
                 "Bacteroides",
                 "Prevotellaceae_UCG.004",
                 "p.251.o5",
                 "dgA.11_gut_group",
                 "Akkermansia",
                 "Alistipes",
                 "Roseburia",
                 "Escherichia.Shigella",
                 "Dorea")
```

Plot more abundant in BF-NP

```{r}
panel1 <- ggplot(df_long_BNP, aes(y = MeanAbundance, x = Site_2, group = Genus))+
  geom_line(aes(col = Genus), size = 1.5)+
  geom_point(aes(fill = Genus), size = 2, col = "black", pch = 21)+
  
  geom_label_repel(data=subset(df_long_BNP, Site_2=="Sumava"), aes(label = bnp_labels, fill = Genus),alpha = 0.5,nudge_x = 1,nudge_y = 0, force=5, fontface="bold",segment.color = 'transparent')+
  
  scale_x_discrete(name = "", labels = c("BF-NP", "S-NP"))+
  ylab("Mean log10 abundance")+
    scale_y_log10(limits = c(1,15000))+
    theme_bw(base_size = 14)+
    theme(legend.position = "none")+
  ggtitle("Genera enriched in BF-NP")
```

Plot more abundant in S-NP

```{r}
panel2 <- ggplot(df_long_SNP, aes(y = MeanAbundance, x = Site_2, group = Genus))+
  geom_line(aes(col = Genus), size = 1.5)+
  geom_point(aes(fill = Genus), size = 2, col = "black", pch = 21)+
  
  geom_label_repel(data=subset(df_long_SNP, Site_2=="Sumava"), aes(label = snp_labels, fill = Genus),alpha = 0.5,nudge_x = 1,nudge_y = 0, force=5, fontface="bold",segment.color = 'transparent')+

  scale_x_discrete(name = "", labels = c("BF-NP", "S-NP"))+
  ylab("Mean log10 abundance")+
  scale_y_log10(limits = c(1,15000))+
  theme_bw(base_size = 14)+
  theme(legend.position = "none")+
  ggtitle("Genera enriched in S-NP")
```

## Venn diagram of sites

Venn for all Genera across sites

```{r}
#install.packages("eulerr") # If not installed
library(eulerr)
library(microbiome)
#devtools::install_github('microsud/microbiomeutilities')
library(microbiomeutilities)

deer_winter_clean <- subset_samples(deer_winter, Site_2 != "NA")
deer_winter_clean

deer_winter_clean <- prune_taxa(taxa_sums(deer_winter_clean) > 0, deer_winter_clean) 

# on genus level 
deer_winter_genus <- tax_glom(deer_winter_clean, taxrank = "Genus")

# simple way to count number of samples in each group
table(meta(deer_winter_genus)$Site_2, useNA = "always")

# convert to relative abundances
pseq.rel <- microbiome::transform(deer_winter_genus, "compositional")


#saveRDS(pseq.rel,"pseq.rel.RDS")

# make a list of sites

sites <- unique(as.character(meta(pseq.rel)$Site_2))
print(sites)


# Now if we would want to plot a venn of that, too, adapt code from venn_core:

list_all <- c() # an empty object to store information

for (n in sites){ # for each variable n in sites
  #print(paste0("Identifying Core Taxa for ", n))
  
  ps.sub <- subset_samples(pseq.rel, Site_2 == n) # Choose sample from feeding by n
  
  core_m <- core_members(ps.sub, 
                         detection = 0/100, 
                         prevalence = 0/100, include.lowest = FALSE)
  print(paste0("No. of taxa in ", n, " : ", length(core_m))) # print core taxa identified in each feeding group.
  list_all[[n]] <- core_m # add to a list core taxa for each group.
  #print(list_core)
}


# plot venn 

str(list_all)

names(list_all)[1] <- "BF-NP"
names(list_all)[2] <- "N-FE"
names(list_all)[3] <- "S-NP"

plot(venn(list_all), fills = c("#3D77A4","#A3CBE1","#A2E086"))

#print(list_all)
```

Add taxa names

```{r}
# use the pseq.rel object created at the begening of this tutorial. 
taxa_names(pseq.rel)[1:5]

# format names
pseq.rel.f <- format_to_besthit(pseq.rel)
# check names
taxa_names(pseq.rel.f)[1:5]


list_all <- c() # an empty object to store information

for (n in sites){ # for each variable n in sites
  #print(paste0("Identifying Core Taxa for ", n))
  
  ps.sub <- subset_samples(pseq.rel.f,  Site_2 == n) 
  
  core_m <- core_members(ps.sub, 
                         detection = 0/100, 
                         prevalence = 0/100, include.lowest = FALSE)
  print(paste0("No. of taxa in ", n, " : ", length(core_m))) # print core taxa identified in each feeding group.
  list_all[[n]] <- core_m # add to a list core taxa for each group.
  #print(list_core)
}


print(list_all)
```

```{r}
sumava_df <- list_all$Sumava
sumava_df = data.frame(sumava_df)
#write.csv(sumava_df, "sumava_df.csv")

neureichenau_df <- list_all$Neureichenau
neureichenau_df = data.frame(neureichenau_df)
#write.csv(neureichenau_df, "neureichenau_df.csv")

bf_df <- list_all$Bavarian_Forest
bf_df = data.frame(bf_df)
#write.csv(bf_df, "bf_df.csv")
```

Venn for shared core ASVs (50% prev.) across sites

```{r}
# Write a for loop to go through each of the sites one by one and combine identified core taxa into a list.

list_core <- c() # an empty object to store information

for (n in sites){ # for each variable n in sites
  #print(paste0("Identifying Core Taxa for ", n))
  
  ps.sub <- subset_samples(pseq.rel, Site_2 == n) # Choose sample from sites by n
  
  core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                         detection = 0, # 0.001 in at least 50% samples 
                         prevalence = 0.5)
  print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each feeding preference
  list_core[[n]] <- core_m # add to a list core taxa for each group.
  #print(list_core)
}

#str(list_core)


# plot venn 

names(list_core)[1] <- "BF-NP"
names(list_core)[2] <- "N-FE"
names(list_core)[3] <- "S-NP"


plot(venn(list_core), fills = c("#3D77A4","#A3CBE1","#A2E086"))

#print(list_core) # this is not really helpful, let's have a look at the taxonomy of these ASVs
```

Find and assign names to core taxa

```{r}
# use the pseq.rel object created at the begening of this tutorial. 
taxa_names(pseq.rel)[1:5]

# format names
pseq.rel.f <- format_to_besthit(pseq.rel)
# check names
taxa_names(pseq.rel.f)[1:5]



list_core <- c() # an empty object to store information

for (n in sites){ # for each variable n in DiseaseState
  #print(paste0("Identifying Core Taxa for ", n))
  
  ps.sub <- subset_samples(pseq.rel.f,  Site_2 == n) # Choose sample from DiseaseState by n
  
  core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                         detection = 0,  
                         prevalence = 0.5)
  print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each DiseaseState.
  list_core[[n]] <- core_m # add to a list core taxa for each group.
  #print(list_core)
}

#print(list_core)
```

```{r}
sumava_df <- list_core$Sumava
sumava_df = data.frame(sumava_df)
#write.csv(sumava_df, "sumava_df50.csv")

neureichenau_df <- list_core$Neureichenau
neureichenau_df = data.frame(neureichenau_df)
#write.csv(neureichenau_df, "neureichenau_df50.csv")

bf_df <- list_core$Bavarian_Forest
bf_df = data.frame(bf_df)
#write.csv(bf_df, "bf_df50.csv")
```

# Infected individuals

Now we focus on the infected individuals - for this we will remove calves.

```{r}
df_winter # this is the df from earlier including all individuals

df_winter_ii <- subset(df_winter, Fluke == "1")
df_winter_ii <- subset(df_winter_ii, Age_Estimate != "Calf")

df_winter_ii # 57 indidivuals
```

F Count among infected individuals

```{r}
str(df_winter_ii$F_Count)

tbl <- with(df_winter_ii, table(F_Count))

ggplot(as.data.frame(tbl), aes(factor(F_Count), Freq, fill = F_Count)) +     
  geom_bar(stat="identity")+
  xlab("Fluke count")+
  ylab("Individuals [n]")+
  theme_classic(base_size=14)
```

Due to the live cycle of F. magna we expect differences between individuals carrying only 1 fluke and those with a high parasite burden (>9). 

To compare them create a categorical variable to identify individuals with one and many (>9) fluke infections

```{r}
deer_winter_ii <- subset_samples(deer_winter, Fluke=="1")
deer_winter_ii <- subset_samples(deer_winter_ii, Site_2 != "NA")
deer_winter_ii = prune_taxa(taxa_sums(deer_winter_ii) > 0, deer_winter_ii)

# add fluke intensity to the sample data so we can filter 
sample_data(deer_winter_ii) <- data.frame(sample_data(deer_winter_ii)) %>%
  mutate(Fluke_intensity = case_when(F_Count < 2 ~ 'ONE',
                                     F_Count > 9 ~ 'MANY'))

deer_winter_da <- subset_samples(deer_winter_ii, Fluke_intensity != "NA")

deer_winter_da <- subset_samples(deer_winter_da, Age_Estimate != "Calf")

table(sample_data(deer_winter_da)$Fluke_intensity)
#saveRDS(deer_winter_da, "deer_winter_da.RDS")
```

## Alpha diversity

# Alpha diversity graphs

```{r}
sample_data(deer_winter_da)
str(sample_data(deer_winter_da)$Fluke_intensity)

P1 <- ggplot(data=subset(sample_data(deer_winter_da), !is.na(Fluke_intensity)), aes(x = Fluke_intensity, y = Observed, color = Fluke_intensity, fill = Fluke_intensity))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  stat_compare_means(method = "wilcox.test", aes(label = sprintf("p = %5.3f", as.numeric(..p.format..)))) +
  scale_color_manual(labels = c("One", "Many"),values=c("#FFF0F3","#F14747"))+
  scale_fill_manual(labels = c("One", "Many"),values=c("#FFF0F3","#F14747"))+
  geom_point(aes(col = Fluke_intensity))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Number of ASVs")+
  xlab("")+ theme(legend.position = "none")

P2 <- ggplot(data=subset(sample_data(deer_winter_da), !is.na(Fluke_intensity)), aes(x = Fluke_intensity, y = Shannon, color = Fluke_intensity, fill = Fluke_intensity))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  stat_compare_means(method = "wilcox.test", aes(label = sprintf("p = %5.3f", as.numeric(..p.format..)))) +
  scale_color_manual(labels = c("One", "Many"),values=c("#FFF0F3","#F14747"))+
  scale_fill_manual(labels = c("One", "Many"),values=c("#FFF0F3","#F14747"))+
  geom_point(aes(col = Fluke_intensity))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Shannon Index")+
  xlab("")+ theme(legend.position = "none")

P3 <- ggplot(data=subset(sample_data(deer_winter_da), !is.na(Fluke_intensity)), aes(x = Fluke_intensity, y = PD, color = Fluke_intensity, fill = Fluke_intensity))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  stat_compare_means(method = "wilcox.test", aes(label = sprintf("p = %5.3f", as.numeric(..p.format..)))) +
  scale_color_manual(labels = c("One", "Many"),values=c("#FFF0F3","#F14747"))+
  scale_fill_manual(labels = c("One", "Many"),values=c("#FFF0F3","#F14747"))+
  geom_point(aes(col = Fluke_intensity))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Faith's PD")+
  xlab("")+ theme(legend.position = "none")

alpha.fluke <- ggarrange(P1,P2,P3, ncol = 3) + 
  theme_classic()
```


```{r}
str(sample_data(deer_winter_da)$Age_Estimate)

P1 <- ggplot(sample_data(deer_winter_da), aes(x = Age_Estimate, y = Observed, color = Age_Estimate, fill = Age_Estimate))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(name = "Age", labels = c("A", "SA"), values=c("#A55C02","#EA8100"))+
  scale_fill_manual(name = "Age", labels = c("A", "SA"), values=c("#A55C02","#EA8100"))+
  geom_point(aes(col = Age_Estimate))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Number of observed ASVs")+
  xlab("")

P2 <- ggplot(sample_data(deer_winter_da), aes(x = Age_Estimate, y = Shannon, color = Age_Estimate, fill = Age_Estimate))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(name = "Age", labels = c("A", "SA"), values=c("#A55C02","#EA8100"))+
  scale_fill_manual(name = "Age", labels = c("A", "SA"), values=c("#A55C02","#EA8100"))+
  geom_point(aes(col = Age_Estimate))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Shannon Index")+
  xlab("")

P3 <- ggplot(sample_data(deer_winter_da), aes(x = Age_Estimate, y = PD, color = Age_Estimate, fill = Age_Estimate))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(name = "Age", labels = c("A", "SA"), values=c("#A55C02","#EA8100"))+
  scale_fill_manual(name = "Age", labels = c("A", "SA"), values=c("#A55C02","#EA8100"))+
  geom_point(aes(col = Age_Estimate))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Faith's PD")+
  xlab("")

alpha.age <- ggarrange(P1,P2,P3, ncol = 3, common.legend = TRUE, legend = "right") + 
  theme_classic()
```

```{r}
str(sample_data(deer_winter_da)$Sex)

P1 <- ggplot(data=subset(sample_data(deer_winter_da), !is.na(Sex)), aes(x = Sex, y = Observed, color = Sex, fill = Sex))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  scale_fill_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  geom_point(aes(col = Sex))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Number of observed ASVs")+
  xlab("")

P2 <- ggplot(data=subset(sample_data(deer_winter_da), !is.na(Sex)), aes(x = Sex, y = Shannon, color = Sex, fill = Sex))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  scale_fill_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  geom_point(aes(col = Sex))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Shannon Index")+
  xlab("")

P3 <- ggplot(data=subset(sample_data(deer_winter_da), !is.na(Sex)), aes(x = Sex, y = PD, color = Sex, fill = Sex))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  scale_color_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  scale_fill_manual(labels = c("F", "M"),values=c("#F6DE22","#60712F"))+
  geom_point(aes(col = Sex))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Faith's PD")+
  xlab("")

alpha.sex <- ggarrange(P1,P2,P3, ncol = 3, common.legend = TRUE, legend = "right") + 
  theme_classic()
```

```{r}
sample_data(deer_winter_da)$Site_2 <- as.factor(sample_data(deer_winter_da)$Site_2)
str(sample_data(deer_winter_da)$Site_2)
table(sample_data(deer_winter_da)$Site_2)

P1 <- ggplot(sample_data(deer_winter_da), aes(x = Site_2, y = Observed, color = Site_2, fill = Site_2))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = Site_2))+
  scale_color_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  scale_fill_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Number of ASVs")+
  xlab("")+ theme(legend.position = "none")

P2 <- ggplot(sample_data(deer_winter_da), aes(x = Site_2, y = Shannon, color = Site_2, fill = Site_2))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = Site_2))+
  scale_color_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  scale_fill_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Shannon Index")+
  xlab("")+ theme(legend.position = "none")

P3 <- ggplot(sample_data(deer_winter_da), aes(x = Site_2, y = PD, color = Site_2, fill = Site_2))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = Site_2))+
  scale_color_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  scale_fill_manual(name = "Area",labels = c("BF-NP","N-FE","S-NP"), values=c("#3D77A4","#A3CBE1","#A2E086"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Faith's PD")+
  xlab("")+ theme(legend.position = "none")

alpha.sites <- ggarrange(P1,P2,P3, ncol = 3) + 
  theme_classic()
```

```{r}
str(sample_data(deer_winter_da)$HuntingMonth)
table(sample_data(deer_winter_da)$HuntingMonth)

# c("#004aad","#0d00a4", "#22007c", "#140152", "#04052e", "#02010a") # different color vector

P1 <- ggplot(sample_data(deer_winter_da), aes(x = HuntingMonth, y = Observed, color = HuntingMonth, fill = HuntingMonth))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = HuntingMonth))+
  scale_color_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  scale_fill_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Number of observed ASVs")+
  xlab("")

P2 <- ggplot(sample_data(deer_winter_da), aes(x = HuntingMonth, y = Shannon, color = HuntingMonth, fill = HuntingMonth))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = HuntingMonth))+
  scale_color_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  scale_fill_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Shannon Index")+
  xlab("")

P3 <- ggplot(sample_data(deer_winter_da), aes(x = HuntingMonth, y = PD, color = HuntingMonth, fill = HuntingMonth))+
  geom_boxplot(alpha=0.7)+theme_classic(base_size=14)+
  geom_point(aes(col = HuntingMonth))+
  scale_color_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  scale_fill_manual(values=c("#EFC0CF","#CB4B73","#C17E93","#A7506B","#603140","#8B0C34"))+
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())+
  ylab("Faith's PD")+
  xlab("")

alpha.months <- ggarrange(P1,P2,P3, ncol = 3, common.legend = TRUE, legend = "right") + 
  theme_classic()
```

## Alpha diversity stats 

```{r}
deer_winter_da_df <- data.frame(sample_data(deer_winter_da))

deer_winter_da_df <- subset(deer_winter_da_df, Sex != "NA")
```

```{r}
m1 <- lm(log10(Observed) ~ Sex + Site_2 + Age_Estimate + Fluke_intensity + HuntingMonth + Seq_depth, data = deer_winter_da_df)
summary(m1)

options(na.action=na.fail)
dredge(m1)

summary(lm(log10(Observed) ~ Site_2 + Age_Estimate + Fluke_intensity + Seq_depth, data = deer_winter_da_df))
```

```{r}
m1 <- lm(Shannon ~ Sex + Site_2 + Age_Estimate + Fluke_intensity + HuntingMonth + Seq_depth, data = deer_winter_da_df)
summary(m1)

options(na.action=na.fail)
dredge(m1)

summary(lm(Shannon ~ Age_Estimate + Site_2 + Seq_depth, data = deer_winter_da_df))
```

```{r}
m1 <- glm(PD ~ Sex + Site_2 + Age_Estimate + Fluke_intensity + HuntingMonth + Seq_depth, data = deer_winter_da_df, family = Gamma("inverse"))
summary(m1)

options(na.action=na.fail)
dredge(m1)

summary(glm(PD ~ Age_Estimate + Seq_depth, data = deer_winter_da_df, family = Gamma("inverse")))
```

## Differential abundance 

```{r}
#deer_winter_da <-readRDS("deer_winter_da.RDS")

# agglomerate to Genus level
deer_winter_da_genus <- tax_glom(deer_winter_da, taxrank = "Genus")

deer_winter_da_df <- data.frame(tax_table(deer_winter_da_genus)[,"Genus"])


## Aldex

# Generate Monte Carlo samples of the Dirichlet distribution for each sample.
# Convert each instance using the centred log-ratio transform.
# This is the input for all further analyses.

x_Fluke_intensity <- aldex.clr(
  reads = otu_table(deer_winter_da_genus),
  conds = sample_data(deer_winter_da_genus)$Fluke_intensity, 
  # 128 recommened for ttest, 1000 for rigorous effect size calculation
  mc.samples = 128, 
  denom = "all",
  verbose = FALSE)


# calculates expected values of the Welch's t-test and Wilcoxon rank test on
# the data returned by aldex.clr

tt_Fluke_intensity <- aldex.ttest(x_Fluke_intensity, paired.test = FALSE, verbose = FALSE)


# determines the median clr abundance of the feature in all samples and in
# groups, the median difference between the two groups, the median variation
# within each group and the effect size, which is the median of the ratio
# of the between group difference and the larger of the variance within groups

effect_Fluke_intensity <- aldex.effect(x_Fluke_intensity, CI = TRUE, verbose = FALSE)


# combine all outputs 
aldex_out_Fluke_intensity <- data.frame(tt_Fluke_intensity, effect_Fluke_intensity)

rownames(aldex_out_Fluke_intensity)
rownames(deer_winter_da_df) 

df_merge <- merge(aldex_out_Fluke_intensity, deer_winter_da_df,
                  by = 'row.names', all = TRUE)


aldex_out_Fluke_intensity <- df_merge

View(aldex_out_Fluke_intensity)

##### plot whichever ####

par(mfrow = c(1, 2))
aldex.plot(
  aldex_out_Fluke_intensity, 
  type = "MA", 
  test = "welch", 
  xlab = "Log-ratio abundance",
  ylab = "Difference",
  cutoff = 0.05
)

aldex.plot(
  aldex_out_Fluke_intensity, 
  type = "MW", 
  test = "welch",
  xlab = "Dispersion",
  ylab = "Difference",
  cutoff = 0.05
)

##### read out results #####

sig.subset<-rownames_to_column(aldex_out_Fluke_intensity, "genus") %>%
  filter(wi.eBH <= 0.05)
```

Create a volcano plot showing the effect sizes and p-values for all taxa

```{r}
par(mfrow = c(1, 1))

plot(aldex_out_Fluke_intensity$effect, aldex_out_Fluke_intensity$wi.eBH, log="y", cex=0.7,
     pch=19, xlab="Effect size", ylab="Wilcoxon Rank Sum test p-value", main="One vs. many flukes")+
  abline(h=0.05, lty=2, col="grey")
```

## Venn of shared / different Genera

All genera

```{r}
deer_winter_da <- prune_taxa(taxa_sums(deer_winter_da) > 0, deer_winter_da) 

# simple way to count number of samples in each group
table(meta(deer_winter_da)$Fluke_intensity, useNA = "always")

# on genus level 
deer_winter_da_genus <- tax_glom(deer_winter_da, taxrank = "Genus")

# convert to relative abundances
pseq.rel <- microbiome::transform(deer_winter_da_genus, "compositional")

# make a list of sites
infection_int <- unique(as.character(meta(pseq.rel)$Fluke_intensity))
#print(infection_int)


# Now if we would want to plot a venn of that, too, adapt code from venn_core:

list_infected <- c() # an empty object to store information

for (n in infection_int){ # for each variable n in sites
  
  ps.sub <- subset_samples(pseq.rel,  Fluke_intensity == n) # Choose sample from infection intensity by n
  core_m <- core_members(ps.sub, 
                         detection = 0/100, 
                         prevalence = 0/100, include.lowest = FALSE)
  print(paste0("No. of taxa in ", n, " : ", length(core_m))) # print core taxa identified in each  group.
  list_infected[[n]] <- core_m # add to a list core taxa for each group.
  #print(list_infected)
}

str(list_infected)

# rename
names(list_infected)[1] <- "One
fluke"  
names(list_infected)[2] <- "Many
flukes"  

# check it worked
str(list_infected) 

# plot venn 
plot(venn(list_infected), fills =c("#FFF0F3", "#F14747"))
```

Add taxa names

```{r}
taxa_names(pseq.rel)[1:5]

# format names
pseq.rel.f <- format_to_besthit(pseq.rel)

# check names
taxa_names(pseq.rel.f)[1:5]


list_infected <- c() # an empty object to store information

for (n in infection_int){ # for each variable n in DiseaseState
  #print(paste0("Identifying Core Taxa for ", n))
  
  ps.sub <- subset_samples(pseq.rel.f,  Fluke_intensity == n) # Choose sample from infection intensity by n
  core_m <- core_members(ps.sub, 
                         detection = 0/100, 
                         prevalence = 0/100, include.lowest = FALSE)
  print(paste0("No. of taxa in ", n, " : ", length(core_m))) # print core taxa identified in each  group.
  list_infected[[n]] <- core_m # add to a list core taxa for each group.
}

#print(list_infected)
str(list_infected)
```

Save results

```{r}
unique_one_df <- list_infected$ONE
unique_one_df = data.frame(unique_one_df)
#write.csv(unique_one_df, "unique_one_df.csv")

unique_many_df <- list_infected$MANY
unique_many_df = data.frame(unique_many_df)
#write.csv(unique_many_df, "unique_many_df.csv")
```

## Venn of shared / different Genera at 50% prevalence

```{r}
list_infected <- c() # an empty object to store information

for (n in infection_int){ # for each variable n in sites
  
  ps.sub <- subset_samples(pseq.rel,  Fluke_intensity == n) # Choose sample from infection intensity by n
  core_m <- core_members(ps.sub, 
                         detection = 0, 
                         prevalence = 0.5, include.lowest = FALSE)
  print(paste0("No. of taxa in ", n, " : ", length(core_m))) # print core taxa identified in each  group.
  list_infected[[n]] <- core_m # add to a list core taxa for each group.
  #print(list_infected)
}


# make a list of infection intensities
str(list_infected)

# rename
names(list_infected)[1] <- "One
fluke"  
names(list_infected)[2] <- "Many
flukes"  

str(list_infected) # check it worked


# plot venn 
plot(venn(list_infected), fills =c("#FFF0F3", "#F14747"))
```

Assign names

```{r}
list_infected <- c() # an empty object to store information

for (n in infection_int){ # for each variable n in DiseaseState
  #print(paste0("Identifying Core Taxa for ", n))
  
  ps.sub <- subset_samples(pseq.rel.f,  Fluke_intensity == n) # Choose sample from infection intensity by n
  core_m <- core_members(ps.sub, 
                         detection = 0, 
                         prevalence = 0.5, include.lowest = FALSE)
  print(paste0("No. of taxa in ", n, " : ", length(core_m))) # print core taxa identified in each  group.
  list_infected[[n]] <- core_m # add to a list core taxa for each group.
}

#print(list_infected)
str(list_infected)
```

```{r}
unique_one_df50 <- list_infected$ONE
unique_one_df50 = data.frame(unique_one_df50)
#write.csv(unique_one_df50, "unique_one_df50.csv")

unique_many_df50 <- list_infected$MANY
unique_many_df50 = data.frame(unique_many_df50)
#write.csv(unique_many_df50, "unique_many_df50.csv")
```

## Relative abundance barplots of deer with many vs. one fluke

```{r}
#deer_winter_da <- readRDS("deer_winter_da.RDS") # the phyloseq object with 
```

Create a color vector

```{r}
colorlist <- c("#a35154",
               "#45c558",
               "#7b56d4",
               "#72be3e",
               "#d98ae3",
               "#42972e",
               "#9b40b5",
               "#a5bb36",
               "#3959cf",
               "#c5af3c",
               "#4884f7",
               "#dc9e3a",
               "#6476e0",
               "#dd7f29",
               "#664db1",
               "#3ac37d",
               "#e348b6",
               "#2b7e30",
               "#b62a99",
               "#81be6d",
               "#d369d0",
               "#709633",
               "#956dcf",
               "#738526",
               "#c75450",
               "#4c6a18",
               "#b96eee",
               "#e63b78",
               "#3ad0b9",
               "#db3750",
               "#41cce7",
               "#b6381f",
               "#5d94e1",
               "#e45f37",
               "#4d965c",
               "#3568b0",
               "#8d7e1f",
               "#744ba0",
               "#6dc18e",
               "#dd498f",
               "#2b6e3f",
               "#e26eb3",
               "#527b3c",
               "#a73883",
               "#40a184",
               "#b73b63",
               "#80cbad",
               "#81498c",
               "#b1bc75",
               "#5d61b0",
               "#a75d23",
               "#50a2d2",
               "#a58dee",
               "#8a4262",
               "#3aadb2",
               "#e5916d",
               "#1a6447",
               "#e78394",
               "#287e63",
               "#a266a6",
               "#8d9853",
               "#ab9ee2",
               "#5d6426",
               "#d68ec0",
               "#7a5c1f",
               "#6869a0",
               "#d0ab6f",
               "#9f7c40",
               "#ac5e83",
               "#9d5939")
```

Family level

```{r}
deer_family_all <- deer_winter_da %>%
  tax_glom(taxrank = "Family") %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt()  %>%
  dplyr::filter(Abundance > 0.01) %>% # at least abundance of at least 1%
  dplyr::arrange(Phylum)
```

```{r}
table(sample_data(deer_winter_da)$Fluke_intensity)
str(sample_data(deer_winter_da)$Fluke_intensity)
sample_data(deer_winter_da)$Fluke_intensity <- factor((sample_data(deer_winter_da)$Fluke_intensity), levels = c("ONE","MANY"))
```

By sample ID

```{r}
family_barplot <- ggplot(deer_family_all, aes(x = Sample_ID, y = Abundance, fill = Family))+
  geom_bar(stat = "identity", position = "fill")+
  scale_fill_manual(values=colorlist)+
  guides(fill=guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1, size = 16))+
  ylab("Relative Abundance")+
  xlab("Sample")+
  guides(fill=guide_legend(title="Family")) + 
  theme(axis.title.x = element_text(size = 14)) + 
  theme(axis.title.y = element_text(size = 14)) + 
  theme_bw() + 
  theme(legend.title = element_text(size = 13))+ 
  theme(text = element_text(size=13))+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+ 
  ggtitle("Bacterial family composition with minimum abundance of 1%")+
  facet_wrap(~Fluke_intensity, scales="free_x")
```

By group

```{r}
family_barplot2 <- ggplot(deer_family_all, aes(x = Fluke_intensity, y = Abundance, fill = Family))+
  geom_bar(stat = "identity", position = "fill")+
  scale_fill_manual(values=colorlist)+
  guides(fill=guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1, size = 16))+
  ylab("Relative Abundance")+
  xlab("Sample")+
  guides(fill=guide_legend(title="Genus")) + 
  theme(axis.title.x = element_text(size = 14)) + 
  theme(axis.title.y = element_text(size = 14)) + 
  theme_bw() + 
  theme(legend.title = element_text(size = 13))+ 
  theme(text = element_text(size=13))+
  ggtitle("Bacterial genus composition with minimum abundance of 1%")
```

Genus level

```{r}
deer_genus_all <- deer_winter_da %>%
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt()  %>%
  dplyr::filter(Abundance > 0.01) %>% # at least abundance of at least 1%
  dplyr::arrange(Phylum)
```

By sample ID

```{r}
genus_barplot <- ggplot(deer_genus_all, aes(x = Sample_ID, y = Abundance, fill = Genus))+
  geom_bar(stat = "identity", position = "fill")+
  scale_fill_manual(values=colorlist)+
  ylab("Relative Abundance")+
  xlab(" ")+
  theme(axis.title.x = element_text(size = 14)) + 
  theme(axis.title.y = element_text(size = 14)) + 
  theme_bw() + 
  theme(text = element_text(size=13))+
  theme(axis.text.x = element_text(angle = 45, hjust = 1), legend.position="none",strip.background =element_rect(fill="white"))+ 
  facet_wrap(~Fluke_intensity, scales="free_x")
```

By group

```{r}
genus_barplot2 <- ggplot(deer_genus_all, aes(x = Fluke_intensity, y = Abundance, fill = Genus))+
  geom_bar(stat = "identity", position = "fill")+
  scale_fill_manual(values=colorlist)+
  guides(fill=guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1, size = 16))+
  ylab("Relative Abundance")+
  xlab(" ")+
  guides(fill=guide_legend(title="Genus")) + 
  theme(axis.title.x = element_text(size = 14)) + 
  theme(axis.title.y = element_text(size = 14)) + 
  theme_bw() + 
  theme(legend.title = element_text(size = 13))+ 
  theme(text = element_text(size=13))

## Based on these barplots have a look at specific taxa with Wilcoxon tests
```
